day3/part1.py

```diff
@@ -0,0 +1,37 @@
+if __name__ == "__main__":
+    
+    # Read text file into numbers within an array
+    with open("day3/input.txt") as values:
+         binary_input = values.readlines()
+
+    # Why is len(binary_input[0]) = 6?
+    col = len(binary_input[0]) - 1   

```

Comment:
Also this could fail if `binary_input` is an empty list so I would usually just calculate the "width" in each for loop iteration based on the current array element you've accessed.

For instance:

```python
for i in range(len(array_2d)):
    row = array_2d[i]
    for j in range(len(row)):
        element = row[j]
```

Reply:
I don't think this will work... I need to access the length of elements in each row in the outer loop, not the inner one

Reply:
The problem is that I need to access the width of the array first, so I can't have a variable as the element to access to find the length of each row.

for i in range(len(array_2d[0])):
    for j in range(len(array_2d));
        element = row[i][j]